lib/drivers/medtronic600Driver.js

```diff
@@ -0,0 +1,1418 @@
+      RECEIVE_MESSAGE: 0x80,
+      SEND_MESSAGE_RESPONSE: 0x81,
+      REQUEST_LINK_KEY_RESPONSE: 0x86,
+    };
+  }
+
+  static get ENVELOPE_SIZE() {
+    return 33;
+  }
+
+  static get MINIMED_HEADER() {
+    return 0x5103; // Q\x03
+  }
+
+  static oneByteChecksum(buffer) {
+    // eslint-disable-next-line no-bitwise
+    return _.reduce(buffer, (a, b) => a + b, 0) & 0xff;
+  }
+}
+
+class MinimedResponse extends MinimedMessage {
+  constructor(payload) {
+    super(payload);
+
+    if (this.payload.readUInt16BE(0x00) !== MinimedMessage.MINIMED_HEADER) {
+      throw new InvalidMessageError('Unexpected MiniMed packet header.');
+    }
+
+    const minimedPayloadSize = this.payload.readUInt16LE(0x1C);
+    // Check the payload's checksums
+    const minimedPayload =
+      Buffer.from(this.payload.slice(0x00, MinimedMessage.ENVELOPE_SIZE + minimedPayloadSize));
+    const expectedChecksum = minimedPayload[0x20];
+    const calculatedChecksum =
+      // eslint-disable-next-line no-bitwise
+      (MinimedMessage.oneByteChecksum(minimedPayload) - expectedChecksum) & 0xFF;
+
+    if (expectedChecksum !== calculatedChecksum) {
+      throw new ChecksumError(expectedChecksum, calculatedChecksum);
+    }
+  }
+}
+
+class MinimedRequest extends MinimedMessage {
+  constructor(commandType, pumpSession, payload, responseType = MinimedResponse) {
+    if (new.target === MinimedRequest) {
+      throw new TypeError(`Cannot construct ${new.target.name} instances directly`);
+    }
+
+    super(MinimedRequest.buildPayload(commandType, pumpSession, payload), responseType);
+    this.pumpSession = pumpSession;
+  }
+
+  static buildPayload(commandType, pumpSession, payload) {
+    const payloadLength = (payload === null) ? 0 : payload.length;
+    const payloadBuffer = Buffer.alloc(MinimedMessage.ENVELOPE_SIZE + payloadLength);
+
+    /* eslint-disable lodash/prefer-lodash-method */
+    payloadBuffer.writeUInt16BE(MinimedMessage.MINIMED_HEADER, 0); // Q\x03
+    payloadBuffer.write('000000', 2); // Pump serial. '000000' for 600-series
+    payloadBuffer.fill(0, 8, 18); // Unknown bytes (hardcoded in CLPro)
+    payloadBuffer.writeUInt8(commandType, 18);
+    payloadBuffer.writeUInt32LE(pumpSession.envelopeSequenceNumber += 1, 19);
+    payloadBuffer.fill(0, 23, 28); // Unknown bytes (hardcoded in CLPro)
+    payloadBuffer.writeUInt16LE(payloadLength, 28);
+    payloadBuffer.fill(0, 30, 32); // Unknown bytes (hardcoded in CLPro)
+    payloadBuffer.writeUInt8(0, 32); // Placeholder for the single byte checksum
+    if (payloadLength > 0) {
+      payloadBuffer.write(Buffer.from(payload).toString('binary'), 33, 'binary');
+    }
+    /* eslint-enable lodash/prefer-lodash-method */
+
+    // Now that we have written the message, calculate the CRC
+    const checksum = MinimedMessage.oneByteChecksum(payloadBuffer);
+    payloadBuffer.writeUInt8(checksum, 32);
+    return payloadBuffer;
+  }
+}
+
+class TelDMessage extends MinimedMessage {
```

Comment:
From the CLPro code. The newer protocol messages are identified as `TelD`, and the encrypted sections of the message as `ComD`.